### PR TITLE
ADEN-13924 - upgrade on tvguide mtc ver

### PR DESCRIPTION
**Pull request by adi-wesolowski-fandom**
- call new custom function
- reduce size of package a little + clean up

### CODE REVIEW

File: src/platforms/news-and-ratings/tvguide-mtc/setup/MtcAdsMode.ts [hunk 1]
```diff
@@ -44,4 +46,11 @@ export class MtcAdsMode implements DiProcess {
 			true,
 		);
 	}
+	private prepareCallingExternalFunctionOnUap() {
```
Comment (Jorewin) [inline src/platforms/news-and-ratings/tvguide-mtc/setup/MtcAdsMode.ts:49]: nitpick: I think we could benefit with renaming this to something that reads more like a sentence, e.g. `setupCustomAdLoaderCall` or if we want to stay with an iteration of what we have then: `prepareCallToExternalFunctionOnUap`

File: src/platforms/news-and-ratings/tvguide-mtc/setup/tracking/tracking.setup.ts [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+			({ eventInfo }) => {
+				this.dwTracker.track(eventInfo, trackingUrls.AD_ENG_PLAYER_INFO);
+			},
+			false,
+		);
+
+		porvataTracker.register();
+	}
+
+	// candidate to remove
```
Comment (Jorewin) [inline src/platforms/news-and-ratings/tvguide-mtc/setup/tracking/tracking.setup.ts:84]: If attaching comments, please be more expressive and if these are only notes for yourself, then perhaps we could consider not attaching them at all.

File: src/ad-products/templates/uap/uap-video-settings.ts [hunk 3]
```diff
@@ -1,27 +0,0 @@
-import { resolvedState } from './resolved-state';
-import { UapParams } from './universal-ad-package';
-
-export class UapVideoSettings {
```
[the remark this answers is not in the document]
Comment (adi-wesolowski-fandom) [reply src/ad-products/templates/uap/uap-video-settings.ts:4]: thx

File: src/communication/event-types.ts [hunk 4]
```diff
@@ -146,6 +147,10 @@ export const eventsRepository = {
 	AD_ENGINE_UAP_NTC_LOADED: {
 		name: 'UAP NTC loaded',
 	},
+	AD_ENGINE_CUSTOM_AD_LOADER_CALLED: {
+		name: 'Custom Ad Loader',
```
[the remark this answers is not in the document]
Comment (adi-wesolowski-fandom) [reply src/communication/event-types.ts:151]: ok

File: src/platforms/news-and-ratings/tvguide-mtc/setup/MtcAdsMode.ts (hunk 1 above)
Comment (adi-wesolowski-fandom) [reply src/platforms/news-and-ratings/tvguide-mtc/setup/MtcAdsMode.ts:49]: ok

File: src/platforms/news-and-ratings/tvguide-mtc/setup/tracking/tracking.setup.ts
[the remark this answers is not in the document]
Comment (adi-wesolowski-fandom) [reply src/platforms/news-and-ratings/tvguide-mtc/setup/tracking/tracking.setup.ts:1]: temporary, as we want to reduce size of the uap package part further. But you are right, some of it is duplicated. I can fix it

File: src/platforms/news-and-ratings/tvguide-mtc/setup/tracking/tracking.setup.ts (hunk 2 above)
Comment (adi-wesolowski-fandom) [reply src/platforms/news-and-ratings/tvguide-mtc/setup/tracking/tracking.setup.ts:84]: no, general comments, but also some leftovers. ok, i'll remove it

File: src/platforms/news-and-ratings/tvguide-mtc/tvguide-mtc-platform.ts [hunk 5]
```diff
@@ -5,15 +5,14 @@ import {
 	InstantConfigSetup,
 	MetricReporterSetup,
 	NoAdsMode,
-	PlatformContextSetup,
```
[the remark this answers is not in the document]
Comment (adi-wesolowski-fandom) [reply src/platforms/news-and-ratings/tvguide-mtc/tvguide-mtc-platform.ts:8]: Screenshot 2024-03-04 at 12 57 16
I see its still used in some places

File: src/platforms/news-and-ratings/tvguide-mtc/setup/tracking/tracking.setup.ts
[the remark this answers is not in the document]
Comment (adi-wesolowski-fandom) [reply src/platforms/news-and-ratings/tvguide-mtc/setup/tracking/tracking.setup.ts:1]: done

File: src/ad-products/templates/uap/interface/uap-config.ts
[the remark this answers is not in the document]
Comment (adi-wesolowski-fandom) [reply src/ad-products/templates/uap/interface/uap-config.ts:1]: I renamed dir, for now. During process of reducing uap-package/part size, I will try to handle it better/cleaner

File: src/platforms/news-and-ratings/tvguide-mtc/setup/tracking/tracking.setup.ts
[the remark this answers is not in the document]
Comment (Jorewin) [reply src/platforms/news-and-ratings/tvguide-mtc/setup/tracking/tracking.setup.ts:1]: Thank you!

**Review by Jorewin — DISMISSED** [review]
Looks good!

**Review by mklakus-fandom — APPROVED** [review]
🚀

**Pull request merged**